clippy_lints/src/collapsible_if.rs

```diff
@@ -163,8 +163,9 @@ impl CollapsibleIf {
         }
     }
 
-    pub fn eligible_condition(&self, cond: &Expr<'_>) -> bool {
-        self.let_chains_enabled || !matches!(cond.kind, ExprKind::Let(..))
+    fn eligible_condition(&self, cx: &LateContext<'_>, cond: &Expr<'_>) -> bool {
+        !matches!(cond.kind, ExprKind::Let(..))
+            || (cx.tcx.sess.edition().at_least_rust_2024() && self.msrv.meets(cx, msrvs::LET_CHAINS))
```

Comment:
Any reason to prefer passing an extra argument every time (`cx`) and doing an edition comparison, while the edition being ≥2024 could instead be checked once from `tcx` at `CollapsibleIf::new()` time? The performance hit should be unnoticeable, but I wonder why you made this choice.

Reply:
`cx` is required for the MSRV check so doing so doesn't remove the extra arg

It could've been done once in `new`, it's not really something I actively considered. Perf wise it's an extra pointer dereference or two, doesn't seem worth splitting up the condition